notification-consumer-services/src/main/java/uk/ac/bbsrc/tgac/miso/notification/consumer/service/mechanism/IlluminaNotificationMessageConsumerMechanism.java

```diff
@@ -167,8 +169,7 @@ public void setAttemptRunPopulation(boolean attemptRunPopulation) {
 
               if (run.has("startDate")) {
                 try {
-                  if (run.get("startDate") != null && !run.getString("startDate").equals("null")
-                      && !"".equals(run.getString("startDate"))) {
+                  if (!"null".equals(run.getString("startDate")) && !isStringEmptyOrNull(run.getString("startDate"))) {
```

Comment:
This one of those fun cases where equals is not commutative because the receiver is special. `"x".equals(null)` is false, while `((String)null).equals("x")` is an exception. I'm happy to reverse it for clarity.